mitodl/ocw-studio
# Add site listing at site dashboard

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #106 
Fixes #107 
Fixes #53 

#### What's this PR do?
Implements the site listing with pagination, and moves the existing add new site UI to its own page at `/new-site/`.

#### How should this be manually tested?
 - Go to `/sites/`. You should see a list of courses with a `>` link at the bottom to go to next page. Click that link and you should see content show "Loading" and then show the next page of results. On the second page you should see a `<` link to go back to the previous page. The page should not do a full refresh but you should see the URL change to include `offset=10` for whatever the offset is expected to be. 
 - There should be an "Add new" link which goes to a form to create a new site at `/new-site/`
 - Links to the site detail pages should work as well.

#### Reviewer notes
`SitesDashboard.tsx` and its test file were renamed to `SiteCreationPage.tsx` with only certain variables renamed like `Website` to `WebsiteDetail`. Then I made a new file also called `SitesDashboard.tsx` which has most of the new functionality for this PR. This means the diff looks strange but in general you can ignore `SiteCreationPage.tsx` and treat `SitesDashboard.tsx` as a brand new file instead of an updated one (and same for the test files).


#### Screenshots (if appropriate)
![Screenshot_2021-03-11 ocw_studio](https://user-images.githubusercontent.com/863262/110798205-ec14b100-8247-11eb-8791-84435af1d161.png)
